pandas/tseries/offsets.py

```diff
@@ -1665,7 +1665,50 @@ class YearBegin(YearOffset):
 # Special Offset Classes
 
 
-class FY5253(SingleConstructorOffset):
+class FY5253Mixin(BaseOffset):
+    def __init__(
+        self, n=1, normalize=False, weekday=0, startingMonth=1, variation="nearest"
+    ):
+        BaseOffset.__init__(self, n, normalize)
+        object.__setattr__(self, "startingMonth", startingMonth)
```

Comment:
question but do we need to reference object here any more or can we not just reference the built in setattr? This is a Py2 compat way of setting things no?

Reply:
> This is a Py2 compat way of setting things no?

No, it is is compat for a recently-removed `def __setattr__` that raised AttributeError to make DateOffsets immutable.  We removed that `__setattr__` patch because it didnt place nicely with making BaseOffset a cdef class.  So we're in a bit of a limbo until we get all of the `__init__` methods to be cdef-class-only.